RxSwift/Observables/RepeatWhen.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  RepeatWhen.swift
+//  RxSwift
+//
+//  Created by sergdort on 12/05/2017.
+//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
+//
+
+extension ObservableType {
+    /**
+    Returns an Observable that emits the same values as the source Observable with the exception of an onCompleted. An onCompleted notification from the source will result in the emission of a void item to the Observable provided as an argument to the notificationHandler function. If that Observable calls onComplete or onError then repeatWhen will call onCompleted or onError on the child subscription. Otherwise, this Observable will resubscribe to the source observable.
+     
+     - parameter notificationHandler: receives an Observable of notifications with which a user can complete or error, aborting the repeat.
+     
+     - returns: An observable sequence producing the elements of the given sequence repeatedly
+    */
+    public func repeatWhen<TriggerObservable: ObservableType>(_ notificationHandler: @escaping (Observable<Void>) -> TriggerObservable)
```

Comment:
Yeah I agree with here, it doesn't really fit within the other implementation or the usage of it :) I think it's good without.